content/docs/command-reference/config.md

```diff
@@ -179,11 +179,11 @@ This section contains the following options, which affect the project's
   > set.
 
 - `cache.shared` - permissions for newly created or downloaded cache files and
-  directories. The default is `0o664`(rw-r--r--) for files and `0o755`
-  (rwxr-xr-x) for directories. The only accepted value right now is `group`,
-  which makes DVC use `0o664` (rw-rw-r--) for files and `0o775` (rwxrwxr-x) for
-  directories, which is useful when you are using a a
-  [shared development server](/doc/use-cases/shared-development-server).
+  directories. The only accepted value right now is `group`, which makes DVC use
+  `0o664` (rw-rw-r--) for files and `0o775` (rwxrwxr-x) for directories, which
+  is useful when you are using a a
+  [shared development server](/doc/use-cases/shared-development-server). (The
+  default permissions for cache files is system dependent.)
```

Comment:
> (The default permissions for cache files is system dependent.)

No need for parentheses I think. But what are the defaults? (`664/755` also seem system-dependent.)

Reply:
Other options have parenthesis while mentioning defaults, so I put to this but both ways are fine with me. 

The defaults are determined by `umask` and `acl` in Linux. (`man umask` has a description and example.) I think it's a distraction at that point in the command reference to describe the default behavior in full detail. If DVC doesn't have a custom cache permissions set, we can remove the part even in the parenthesis.

Reply:
From the [code](https://github.com/iesahin/dvc/blob/08300d3bcf51f7bd9c8509924110416461b11b60/dvc/cache/local.py#L33), it looks the permissions are set by `umask`.

[Here](https://github.com/iesahin/dvc/blob/08300d3bcf51f7bd9c8509924110416461b11b60/dvc/utils/fs.py#L16) it seems that `umask` is set to system default. (What does this mean in Windows, I'm not sure.)

Reply:
No need for parentheses. It's kind of strange to have a full sentence in parenthesis anyway (I probably wrote that so my bad).

But mostly I'm still confused what the defaults are. Isn't `group` the only accepted value? If you don't set `cache.shared`, what happens? Is it even meaningful to talk about defaults for this config option? Thanks

Reply:
p.s. `umask = umask = os.umask(0) = 0o000` per https://github.com/iesahin/dvc/blob/08300d3bcf51f7bd9c8509924110416461b11b60/dvc/utils/fs.py#L16

So I guess `0o666 & ~umask = 0o666 & 0x111` always = 0c666? (always 777 for dirs)

Reply:
[`umask`](https://docs.python.org/3/library/os.html#os.umask) is a strange function. It returns the current `umask` when you set it to something but there is no other way to learn current `umask.` So the code sets `umask` to 0 to get the current value, then set `umask` to the previously obtained value. 

The code uses it to change permissions in [`chmod`](https://docs.python.org/3/library/os.html#os.chmod) but as @efiop said, Windows doesn't support this so it doesn't matter actually. 

For Linux/Unix, it looks `chmod` is able to modify permissions arbitrarily. e.g. When `umask` was `777`, I was able to set a file's permissions to `r--/---/---` with `os.chmod("file", 0o400)`

Reply:
OK. Resolving this long review and reopening a short one in the latest text...